oozie/src/main/java/org/apache/falcon/oozie/process/NativeOozieProcessWorkflowBuilder.java

```diff
@@ -0,0 +1,222 @@
+        Properties props = new Properties();
+
+        if (entity.getInputs() == null) {
+            props.put(WorkflowExecutionArgs.INPUT_FEED_NAMES.getName(), NONE);
+            props.put(WorkflowExecutionArgs.INPUT_FEED_PATHS.getName(), NONE);
+            props.put(WorkflowExecutionArgs.INPUT_NAMES.getName(), NONE);
+            props.put(WorkflowExecutionArgs.INPUT_STORAGE_TYPES.getName(), NONE);
+            return props;
+        }
+        List<String> falconInputFeeds = new ArrayList<>();
+        List<String> falconInputNames = new ArrayList<>();
+        List<String> falconInputPaths = new ArrayList<>();
+        List<String> falconInputFeedStorageTypes = new ArrayList<>();
+        for (Input input : entity.getInputs().getInputs()) {
+            Feed feed = ConfigurationStore.get().get(EntityType.FEED, input.getFeed());
+            Storage storage = FeedHelper.createStorage(clusterObj, feed);
+            if (storage.getType() != Storage.TYPE.FILESYSTEM) {
+                throw new UnsupportedOperationException("Storage Type not supported " + storage.getType());
+            }
+            falconInputFeeds.add(feed.getName());
+            falconInputNames.add(input.getName());
+            falconInputFeedStorageTypes.add(storage.getType().name());
+            String partition  = input.getPartition();
+
+            String startTimeExp = input.getStart();
+            String endTimeExp = input.getEnd();
+            ExpressionHelper.setReferenceDate(new Date(getNominalTime().getMillis()));
+            Date startTime = EXPRESSION_HELPER.evaluate(startTimeExp, Date.class);
+            Date endTime = EXPRESSION_HELPER.evaluate(endTimeExp, Date.class);
+
+            for (org.apache.falcon.entity.v0.feed.Cluster cluster : feed.getClusters().getClusters()) {
+                org.apache.falcon.entity.v0.cluster.Cluster clusterEntity =
+                        EntityUtil.getEntity(EntityType.CLUSTER, cluster.getName());
+                if (!EntityUtil.responsibleFor(clusterEntity.getColo())) {
+                    continue;
+                }
+
+                List<Location> locations = FeedHelper.getLocations(cluster, feed);
+                for (Location loc : locations) {
+                    if (loc.getType() != LocationType.DATA) {
+                        continue;
+                    }
+                    List<String> paths = new ArrayList<>();
+                    List<Date> instanceTimes = EntityUtil.getEntityInstanceTimes(feed, cluster.getName(),
+                            startTime, endTime); // test when startTime and endTime are equal.
+                    for (Date instanceTime : instanceTimes) {
+                        String path = EntityUtil.evaluateDependentPath(loc.getPath(), instanceTime);
+                        if (StringUtils.isNotBlank(partition)) {
+                            if (!path.endsWith("/") && !partition.startsWith("/")) {
```

Comment:
We are just adding partition to feed instance path. While adding partition, it should be prepend with "/" right ?